src/main/java/de/chojo/repbot/ReputationBot.java

```diff
@@ -193,8 +200,31 @@ private void initBot() {
                     if (roleById == null) return false;
                     return wrapper.getMember().getRoles().contains(roleById);
                 })
-                .withCommandErrorHandler((command, message, throwable) -> {
-                    log.error(LogNotify.NOTIFY_ADMIN, "Command execution of {} failed\n{}", command.command(), message, throwable);
+                .withCommandErrorHandler((context, throwable) -> {
+                    if (throwable instanceof InsufficientPermissionException) {
+                        var permissionException = (InsufficientPermissionException) throwable;
+                        var permission = permissionException.getPermission();
+                        var errorMessage = localizer.localize("error.missingPermission", context.guild(),
+                                Replacement.create("PERM", permission.getName(), Format.BOLD));
+                        if (context.guild().getSelfMember().hasPermission(permission)) {
+                            errorMessage += "\n" + localizer.localize("error.missingPermissionChannel", context.guild(),
+                                    Replacement.createMention((TextChannel) context.channel()));
+                        } else {
+                            errorMessage += "\n" + localizer.localize("error.missingPermissionGuild", context.guild());
+                        }
+                        if (permissionException.getPermission() != Permission.MESSAGE_WRITE) {
+                            context.channel().sendMessage(errorMessage).queue();
+                            return;
+                        }
+                        var ownerId = context.guild().getOwnerIdLong();
+                        var finalErrorMessage = errorMessage;
+                        context.guild().retrieveMemberById(ownerId)
+                                .queue(member -> member.getUser().openPrivateChannel()
+                                        .queue(privateChannel -> privateChannel.sendMessage(finalErrorMessage)
+                                                .queue(succ -> {}, err -> ErrorResponseException.ignore(ErrorResponse.CANNOT_SEND_TO_USER))));
+                        return;
```

Comment:
that's what flatMap and onErrorFlatMap is for

Reply:
```java
context.guild().retrieveMemberById(ownerId)
    .flatMap(member -> member.getUser().openPrivateChannel())
    .flatMap(privateChannel -> privateChannel.sendMessage(finalErrorMessage))
    .onErrorMap(t -> null)
    .queue();
```
Something like this?

Reply:
can't you do the ErrorResponse.ignore thing in the flat map

Reply:
No the onErrorMap needs to return a message. so I can only return null.

Reply:
Fucking hell that's stupid ahhhhhhh.
No afaik it doesn't so either we return null 
Or we use the ugly queue we had before and be more explicit about the ignore

Reply:
the null return still looks better imo. I will keep this and push it.